app/browser/tabs.js

```diff
@@ -619,6 +623,79 @@ const api = {
       api.createTab(state, action)
     }
     return state
+  },
+
+  goBack: (state, action) => {
+    action = makeImmutable(action)
+    const tab = api.getWebContents(action.get('tabId'))
+    if (tab && !tab.isDestroyed()) {
+      tab.goBack()
+    }
+    return state
+  },
+
+  goForward: (state, action) => {
+    action = makeImmutable(action)
+    const tab = api.getWebContents(action.get('tabId'))
+    if (tab && !tab.isDestroyed()) {
+      tab.goForward()
+    }
+    return state
+  },
+
+  goToIndex: (state, action) => {
+    action = makeImmutable(action)
+    const tab = api.getWebContents(action.get('tabId'))
+    if (tab && !tab.isDestroyed()) {
+      tab.goToIndex(action.get('index'))
+    }
+    return state
+  },
+
+  getHistoryEntries: (state, action) => {
+    const tab = api.getWebContents(action.get('tabId'))
+    const sites = state ? state.get('sites') : null
+
+    if (tab && !tab.isDestroyed()) {
+      let history = {
+        count: tab.getEntryCount(),
+        currentIndex: tab.getCurrentEntryIndex(),
+        entries: []
+      }
+
+      for (let index = 0; index < history.count; index++) {
+        const url = tab.getURLAtIndex(index)
+        const title = tab.getTitleAtIndex(index)
+
+        let entry = {
+          index: index,
+          url: url,
+          display: title || url,
+          icon: null
+        }
+
+        if (isTargetAboutUrl(url)) {
+          // TODO: return brave lion (or better: get icon from extension if possible as data URI)
+        } else {
+          if (sites) {
+            const site = sites.find(function (element) { return element.get('location') === url })
+            if (site) {
+              entry.icon = site.get('favicon')
+            }
+          }
+
+          if (!entry.icon) {
+            entry.icon = getDefaultFaviconUrl(url)
```

Comment:
you are correct we are not receiving anything here, that's why I added if typeof undefined